onadata/libs/utils/logger_tools.py

```diff
@@ -196,6 +195,15 @@ def save_attachments(xform, instance, media_files):
             mimetype=content_type,
             extension=extension)
 
+    instance.total_media = len([m for m in instance.get_dict().keys() if m in
```

Comment:
I think this may not handle images in a repeat. the structure is slightly different. I will suggest looping through each and every key in the submission JSON.